bin/switchy

```diff
@@ -0,0 +1,116 @@
+
+import sys
+import re
+import time
+import click
+import switchy
+import logging
+
+@click.group()
+def cli():
+    pass
+
+@cli.command('list-apps')
+def list_apps():
+    switchy.apps.load()
+    click.echo('Registered Switchy Apps:\n')
+    for app in switchy.apps.next():
+        click.echo('{} - {}'.format(app.name, app.help))
+
+@cli.command()
+@click.argument('file-name', nargs=1, required=True, type=click.Path(exists=True))
+def plot(file_name):
+    from switchy.apps.measure import metrics
+    m = metrics.load(file_name)
+    click.echo('Plotting {} ...\n'.format(file_name))
+    m.plot(block=True)
+
+@cli.command()
+@click.argument('slaves', nargs=-1, required=True)
+@click.option('--proxy', default=None, help='Hostname or IP address of the proxy device (this is usually the device you are testing)', required=True)
+@click.option('--profile', default='internal', help='Profile to use for outbound calls in the load slaves')
+@click.option('--rate', default=None, help='Call rate')
+@click.option('--limit', default=None, help='Maximum number of concurrent calls')
+@click.option('--max-offered', default=None, help='Maximum number of calls to place before stopping the program')
+@click.option('--duration', default=None, help='Duration of calls in seconds')
+@click.option('--interactive/--non-interactive', default=False, help='Whether to jump into an interactive session after setting up the call originator')
+@click.option('--debug/--no-debug', default=False, help='Whether to enable debugging')
+@click.option('--app', default='Bert', help='Switchy application to execute (see list-apps command to list available apps)')
+@click.option('--metrics-file', default=None, help='Store metrics at the given file location')
+def run(slaves, proxy, profile, rate, limit, max_offered, duration, interactive, debug, app, metrics_file):
+    logging.basicConfig()
+
+    # Check if the specified (or default) app is valid
+    switchy.apps.load()
+    appdata = switchy.apps.get(app)
+    if not appdata:
+        raise click.ClickException('Unknown app {}. Use list-apps command to list available apps'.format(app))
+
+    # TODO: get_originator() receives an apps tuple (defaults to Bert) to select the application
+    # we should accept --app multi argument list to set multiple apps
+    o = switchy.get_originator(slaves, apps=(appdata.cls, ))
+
+    # Set basic originator settings
+    if rate:
+        o.rate = int(rate)
+    if limit:
+        o.limit = int(limit)
+    if max_offered:
+        o.max_offered = int(max_offered)
+    if duration:
+        o.duration = int(duration)
+
+    # Prepare the originate string for each slave
+    # depending on the profile name and network settings
+    # configured for that profile in that particular slave
+    p = re.compile('.+?BIND-URL\s+?.+?@(.+?):(\d+).+?\s+', re.IGNORECASE | re.DOTALL)
+    for c in o.pool.clients:
+        status = c.client.api('sofia status profile {}'.format(profile)).getBody()
```

Comment:
use `Client.cmd` instead: https://github.com/sangoma/switchy/blob/master/switchy/observe.py#L1121
It has the body access and whitespace stripping built in.